tronbyt_server/api.py

```diff
@@ -34,6 +34,36 @@ def get_api_key_from_headers(headers: Headers) -> Optional[str]:
     return None
 
 
+@bp.route("/devices", methods=["GET"])
+def list_devices() -> ResponseReturnValue:
+    api_key = get_api_key_from_headers(request.headers)
+    if not api_key:
+        abort(
+            HTTPStatus.BAD_REQUEST,
+            description="Missing or invalid Authorization header",
+        )
+
+    user = db.get_user_by_api_key(api_key)
+    if not user:
+        abort(HTTPStatus.UNAUTHORIZED, description="Invalid API key")
+
+    devices = user.get("devices", {})
+    metadata = [
+        {
+            "id": device_id,
```

Comment:
This construction is duplicated in `get_device()` so you might want to extract this to a common function to ensure consistency.